charmcraft/commands/store/__init__.py

```diff
@@ -922,7 +922,10 @@ def _get_lib_info(*, full_name=None, lib_path=None):
         # dir and Python extension.
         #    e.g.: charms.mycharm.v4.foo -> lib/charms/mycharm/v4/foo.py
         try:
-            charmsdir, importable_charm_name, v_api, libfile = full_name.split(".")
+            charmsdir, charm_name, v_api, libfile = full_name.split(".")
+
+            # the lib full_name includes the charm_name which might not be importable (dashes)
+            importable_charm_name = create_importable_name(charm_name)
```

Comment:
This comment and code are fine, but they should be placed outside the try/except structure.

Reply:
Thanks for the review. I updated the commit with the change.